test/grouping.jl

```diff
@@ -3710,6 +3710,14 @@ end
     end
 end
 
+@testset "grouping floats" begin
+    @test length(groupby_checked(DataFrame(a=[0.0, -0.0]), :a)) == 2
+    @test getindex.(keys(groupby_checked(DataFrame(a=[3.0, 2.0, 0.0]), :a)), 1) ==
+          [0, 2, 3]
+    @test getindex.(keys(groupby_checked(DataFrame(a=[3.0, 2.0, -0.0]), :a)), 1) ==
+          [3, 2, 0]
+end
```

Comment:
in the tests above you see the consequence of allowing floats in fast path. The issue is that then the order of groups (being a consequence of switching of algorithm) depends on the values not only on types.

Reply:
Ah, right. We don't provide any guarantees on the order, but it's true that the change could be confusing.

Reply:
Before I move forward with the changes you suggested above let us make this decision:
1. do we want to be flexible (and **sometimes** handle floats using fast path),
2. or we want to use fast path only to `Integer`s?

I prefer option 2. (i.e. do not handle floats using fast path at all), but I also understand the reasons why you would prefer it, so I am not sure which way to go.